### ISSUE TITLE
Any way to send full path along with attached file ?

### ISSUE THREAD

**Issue by cyrilchapon** [visionmedia/superagent#1257]
Using something like 

```
superagent
  .post('/endpoint')
  .attach('file', 'C:/Users/user1/Desktop/file1.txt', 'Desktop/file1.txt');
```

Sends
```
  [ 'form-data; name="file"; filename="file1.txt"' ]
```

Obviously only `base` is taken from the filename.

If I do need to send a path in the filename, how to do it ?

**Comment by pornel** [comment]
In the browser it's forbidden on purpose (`C:\fakepath\` is hardcoded).

In Node I think we're just passing filename as-is, so it should be work.

If the path is important, it may be better to send it as a separate form field.

**Comment by cyrilchapon** [comment]
> In Node I think we're just passing filename as-is, so it should be work.

Nah in fact we're not, because of using [*form-data* 2.1.1](https://github.com/visionmedia/superagent/blob/master/package.json#L31)

This issue was resolved by [this release(2.2.0)](https://github.com/form-data/form-data/releases/tag/v2.2.0).

[Same bug posted on request.js](https://github.com/request/request/issues/2706) btw.

A dependency update would resolve this I guess.

**Comment by MKrausSHC** [comment]
I also just ran into this problem. I'd very much appreciate a solution.
I'm attaching multiple files and it seems they can get reordered until they are parsed on the server side (using express and formidable). Therefore i'm also having issues with supplying the path information with additional fields.
